src/bindings.cc

```diff
@@ -0,0 +1,203 @@
+    }
+}
+
+static void DefineBootstrapInfo(Isolate *isolate, Local<Object> module_obj, GIBaseInfo *info) {
+    GIInfoType type = g_base_info_get_type (info);
+
+    switch (type) {
+    case GI_INFO_TYPE_FUNCTION:
+        DefineFunction (isolate, module_obj, info);
+        break;
+    case GI_INFO_TYPE_OBJECT:
+        DefineObjectFunctions (isolate, module_obj, info);
+        break;
+    case GI_INFO_TYPE_BOXED:
+    case GI_INFO_TYPE_STRUCT:
+        DefineBoxedFunctions (isolate, module_obj, info);
+        break;
+    default:
+        break;
+    }
+}
+
+static void Bootstrap(const FunctionCallbackInfo<Value> &args) {
+    Isolate *isolate = args.GetIsolate ();
+
+    GIRepository *repo = g_irepository_get_default ();
+    GError *error = NULL;
+
+    const char *ns = "GIRepository";
+    g_irepository_require (repo, ns, NULL, (GIRepositoryLoadFlags) 0, &error);
+    if (error) {
+        isolate->ThrowException (Exception::TypeError (String::NewFromUtf8 (isolate, error->message)));
+        return;
+    }
+
+    Local<Object> module_obj = Object::New (isolate);
+
+    int n = g_irepository_get_n_infos (repo, ns);
+    for (int i = 0; i < n; i++) {
+        GIBaseInfo *info = g_irepository_get_info (repo, ns, i);
+        DefineBootstrapInfo (isolate, module_obj, info);
+        g_base_info_unref (info);
+    }
+
+    args.GetReturnValue ().Set (module_obj);
+}
+
+static void GetConstantValue(const FunctionCallbackInfo<Value> &args) {
+    Isolate *isolate = args.GetIsolate ();
+    GIBaseInfo *info = (GIBaseInfo *) BoxedFromWrapper (args[0]);
+    GITypeInfo *type_info = g_constant_info_get_type ((GIConstantInfo *) info);
+    GIArgument garg;
+    g_constant_info_get_value ((GIConstantInfo *) info, &garg);
+    args.GetReturnValue ().Set (GIArgumentToV8 (isolate, type_info, &garg));
+}
+
+static void MakeFunction(const FunctionCallbackInfo<Value> &args) {
+    Isolate *isolate = args.GetIsolate ();
+    GIBaseInfo *info = (GIBaseInfo *) BoxedFromWrapper (args[0]);
+    args.GetReturnValue ().Set (MakeFunction (isolate, info));
+}
+
+static void MakeClass(const FunctionCallbackInfo<Value> &args) {
+    Isolate *isolate = args.GetIsolate ();
+    GIBaseInfo *info = (GIBaseInfo *) BoxedFromWrapper (args[0]);
+    args.GetReturnValue ().Set (MakeClass (isolate, info));
+}
+
+static void MakeBoxed(const FunctionCallbackInfo<Value> &args) {
+    Isolate *isolate = args.GetIsolate ();
+    GIBaseInfo *info = (GIBaseInfo *) BoxedFromWrapper (args[0]);
+    args.GetReturnValue ().Set (MakeBoxed (isolate, info));
+}
+
+static void ObjectPropertyGetter(const FunctionCallbackInfo<Value> &args) {
+    Isolate *isolate = args.GetIsolate ();
+    GObject *gobject = GObjectFromWrapper (args[0]);
+    String::Utf8Value prop_name_v (args[1]->ToString ());
+    const char *prop_name = *prop_name_v;
+
+    GParamSpec *pspec = g_object_class_find_property (G_OBJECT_GET_CLASS (gobject), prop_name);
+    GValue value = { 0, 0 };
```

Comment:
Use `GValue value = G_VALUE_INIT` so you don't depend on the internal structure of `GValue`